### DISCUSSION TITLE
Prefer branches to detached head checkouts

### DISCUSSION THREAD

**Discussion by ghost** [arxanas/git-branchless#776] [Q&A] [answered]
Would it be possible to teach branchless's checkout commands (`next`, `prev` interactive `co`) to prefer a branch checkout rather than detached HEAD if there is a unique branch at the selected commit?

e.g. with this repo/checkout
```
$ git sl
abc (main)
def 
ghi (feature)
$ git rev-parse HEAD
def
$ git branch
(HEAD detached at b0f5945)
main
feature
```
Right now:
  - `git prev` checks out `abc` in detached HEAD (rather than main)
  - `git next` checks out `def` in detached HEAD (rather than the feature branch)
  - `git co` then selecting `abc (main)` checks out `abc` in detached HEAD (rather than main)

In all these cases the selected commit has exactly one branch pointing at it, so it would be possible to unambiguously impute the "correct" branch to checkout instead. 

This behavior would be useful to people like me who use git-branchless in a hybrid mode, taking advantage of `sl`, `restack`, `move`, etc. but still naming and cutting branches to align with PR/branch-based code review systems.

Perhaps this could be done with a preference, and old and/or truly branchless users could keep the existing behavior. 

The two downside of this are the addition of UX complexity:
 - a setting
 - non-obvious behavior in cases where a commit has multiple branches pointing at it (falling back to the existing detatched HEAD checkout might be confusing, but seems like the only option).

**Comment by arxanas** [comment, marked as answer]
It should already do this! See https://github.com/arxanas/git-branchless/discussions/595. You can file a bug if it's not working right.

**Reply by ghost** [reply]
Quite right. Sorry! I was still on 0.5.0.
